exporters/stats_stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExporter.java

```diff
@@ -0,0 +1,269 @@
+  }
+
+  /**
+   * Creates a Stackdriver Stats exporter for an explicit project ID.
+   *
+   * <p>Only one Stackdriver exporter can be created at any point.
+   *
+   * <p>This uses the default application credentials. See {@link
+   * GoogleCredentials#getApplicationDefault}.
+   *
+   * <p>This is equivalent with:
+   *
+   * <pre>{@code
+   * StackdriverExporter.createWithCredentialsAndProjectId(
+   *     GoogleCredentials.getApplicationDefault(), projectId);
+   * }</pre>
+   *
+   * @param projectId the cloud project id.
+   * @param exportInterval the interval to export stats.
+   * @throws IllegalStateException if a Stackdriver exporter is already created.
+   */
+  public static StackdriverExporter createWithProjectId(
+      String projectId, Duration exportInterval) throws IOException {
+    checkState(handler == null, "Stackdriver exporter is already created.");
+    handler = createInternal(null, projectId, exportInterval);
+    return handler;
+  }
+
+  /**
+   * Creates a Stackdriver Stats exporter.
+   *
+   * <p>Only one Stackdriver exporter can be created at any point.
+   *
+   * <p>This uses the default application credentials. See {@link
+   * GoogleCredentials#getApplicationDefault}.
+   *
+   * <p>This uses the default project ID configured see {@link ServiceOptions#getDefaultProjectId}.
+   *
+   * <p>This is equivalent with:
+   *
+   * <pre>{@code
+   * StackdriverExporter.createWithProjectId(ServiceOptions.getDefaultProjectId());
+   * }</pre>
+   *
+   * @param exportInterval the interval to export stats.
+   * @throws IllegalStateException if a Stackdriver exporter is already created.
+   */
+  public static StackdriverExporter create(Duration exportInterval) throws IOException {
+    checkState(handler == null, "Stackdriver exporter is already created.");
+    handler = createInternal(
+        null, ServiceOptions.getDefaultProjectId(), exportInterval);
+    return handler;
+  }
+
+  private static StackdriverExporter createInternal(
+      @Nullable Credentials credentials,
+      String projectId,
+      Duration exportInterval)
+      throws IOException {
+    MetricServiceClient metricServiceClient;
+    if (credentials == null) {
+      metricServiceClient = MetricServiceClient.create();
+    } else {
+      metricServiceClient = MetricServiceClient.create(
+          MetricServiceSettings.newBuilder()
+              .setCredentialsProvider(FixedCredentialsProvider.create(credentials))
+              .build());
+    }
+    return new StackdriverExporter(
+        projectId, metricServiceClient, exportInterval, Stats.getViewManager());
+  }
+
+  /**
+   * Register a {@link View} against this exporter, and upload it as a {@link MetricDescriptor} to
+   * StackDriver.
+   *
+   * @param view the {@code View} to be registered.
+   */
+  public void registerView(View view) {
+    workerThread.registerView(view);
+    // This is required only for custom view definitions. Canonical views should be pre-registered.
```

Comment:
How do we enforce this?

Reply:
We can check if the view is one of [RpcViewConstants](https://github.com/census-instrumentation/opencensus-java/blob/master/core/src/main/java/io/opencensus/stats/RpcViewConstants.java), if so then it doesn't need to be posted to StackDriver. This is not needed for now, since we haven't defined canonical metrics on StackDriver. I've added a TODO here.